source/io/module_javaScriptInvoke.js

```diff
@@ -153,199 +264,39 @@
-                Module.eggShell.dataWriteUInt16(returnValuePointer, returnValue);
-                return false;
-            }
-            return true;
-        };
-
-        var updateUInt32ReturnValue = function (
-            javaScriptReturnTypeName,
-            returnValuePointer,
-            returnValue) {
-            if (javaScriptReturnTypeName === 'number') {
-                Module.eggShell.dataWriteUInt32(returnValuePointer, returnValue);
-                return false;
-            }
-            return true;
-        };
-
-        var updateSingleReturnValue = function (
-            javaScriptReturnTypeName,
-            returnValuePointer,
-            returnValue) {
-            if (javaScriptReturnTypeName === 'number') {
-                Module.eggShell.dataWriteSingle(returnValuePointer, returnValue);
-                return false;
-            }
-            return true;
-        };
-
-        var updateDoubleReturnValue = function (
-            javaScriptReturnTypeName,
-            returnValuePointer,
-            returnValue) {
-            if (javaScriptReturnTypeName === 'number') {
-                Module.eggShell.dataWriteDouble(returnValuePointer, returnValue);
-                return false;
-            }
-            return true;
-        };
-
-        var updateStringReturnValue = function (
-            javaScriptReturnTypeName,
-            returnValuePointer,
-            returnValue) {
-            if (javaScriptReturnTypeName === 'string') {
-                Module.eggShell.dataWriteString(returnValuePointer, returnValue);
-                return false;
-            }
-            return true;
-        };
-
-        var updateBooleanReturnValue = function (
-            javaScriptReturnTypeName,
-            returnValuePointer,
-            returnValue) {
-            if (javaScriptReturnTypeName === 'boolean') {
-                Module.eggShell.dataWriteBoolean(returnValuePointer, returnValue);
-                return false;
-            }
-            return true;
-        };
-
         var updateReturnValue = function (
-            functionNameString,
+            functionName,
             returnPointer,
             returnValue,
             errorStatusPointer,
             errorCodePointer,
             errorSourcePointer) {
-            var newErrorStatus = false;
-            var newErrorCode = ERRORS.NO_ERROR.CODE;
-            var newErrorSource = ERRORS.NO_ERROR.MESSAGE;
+            var returnValueIndex = 0;
+            var returnTypeName = getParameterTypeString(returnPointer, returnValueIndex);
 
-            var javaScriptReturnTypeName = typeof returnValue;
-            var validJavaScriptReturnType =
-                (javaScriptReturnTypeName === 'number') ||
-                (javaScriptReturnTypeName === 'boolean') ||
-                (javaScriptReturnTypeName === 'string') ||
-                (javaScriptReturnTypeName === 'undefined');
-            if (!validJavaScriptReturnType) {
```

Comment:
Please elaborate why this check is no longer needed. The intent was to return an error when the user was returning a JavaScript type we currently don't support.

Reply:
I added this check back.